### ISSUE TITLE
concurrent call failed 

### ISSUE THREAD

**Issue by lchrampupit** [akretion/ooor#91]
Hi,

we are using ruby version 1.9.3 and when we try to call concurrently the ooor_client  we got an exception :
Errno::ENOTSOCK: An operation was attempted on something that is not a socket.
    /apps/Ruby193/lib/ruby/gems/1.9.1/gems/ooor-1.7.1/lib/app/models/ooor_client.rb:17:in `call2'
    /apps/Ruby193/lib/ruby/gems/1.9.1/gems/ooor-1.7.1/lib/app/models/open_object_resource.rb:134:in `rpc_execute_with_all'
    /apps/Ruby193/lib/ruby/gems/1.9.1/gems/ooor-1.7.1/lib/app/models/open_object_resource.rb:291:in `rpc_execute'
    /apps/Ruby193/lib/ruby/gems/1.9.1/gems/ooor-1.7.1/lib/app/models/open_object_resource.rb:338:in `initialize'
    /apps/Ruby193/lib/ruby/gems/1.9.1/gems/ooor-1.7.1/lib/app/models/open_object_resource.rb:48:in `new'
    /apps/Ruby193/lib/ruby/gems/1.9.1/gems/ooor-1.7.1/lib/app/models/open_object_resource.rb:48:in `create'
we do a multi thread unit test to log this information 
like 
 t1=Thread.new { test1(10) }
    t2=Thread.new { test1(10) }
    t1.join
    t2.join
where we call 10 time the method create
do you know how we can avoid that 

Thanks

**Comment by rvalyi** [comment]
Hello @lchrampupit,

this is really unfortunate I couldn't work on this earlier. An other Rails user reported a threading issue and basically told me Ooor would try to write in the same socket in both threads. And from the design it was true.

Ooor had plenty of changes since 2012: mostly it's now designed to use anynomous proxies instead of Class constants for best Rails usage (unless you use the Command Line), it's optimized for caching proxies meta-data between Processes (say Puma server and caching on Redis), it's optimized for having two different anonymous Rails users using the same Ooor Proxy but still different Odoo sessions (e-commerce with anonymous shopping cart). So lot's of changes...

But coming back to thread safety. I did the following test:

```
def test1(n, name)
  n.times do
  ProductProduct.create name: name
  end
end
  
t1=Thread.new { test1(10, "thread1") }
t2=Thread.new { test1(10, "thread2") }
t1.join;t2.join; puts "started"
```

and I also inserted this line in lib/ooor/services.rb at line 113:
```
p "TTTTTTTTTTTTTTTTT", Thread.current.object_id, json_conn.object_id
```

I could see that before my last commit, both threads would use the same connection/socket.

After this commit https://github.com/akretion/ooor/pull/134/files however each thread has its own sockets. 

I think that if you don't do crazy things like changing the proxy schema between threads it should be thread safe now.

Not sure if it will help you but still documenting here for others.
